cEP-0024.md

```diff
@@ -0,0 +1,96 @@
+| cEP      | 24                                                    |
+| Version  | 1.0                                                   |
+| Title    | Implement Gitmate automations and plugins for coala   |
+| Authors  | Vamshi Krishna Bommerla <mailto:bommvams72@gmail.com> |
+| Status   | Proposed                                              |
+| Type     | Process                                               |
+
+## Abstract
+
+This cEP describes the improvements to the coala workflow and the necessary
+automation to realise it, as a part of the
+[GSoC project](https://summerofcode.withgoogle.com/projects/#5344316227059712).
+
+## Introduction
+
+[GitMate](https://gitmate.io/) is an automation tool for developers. While it
+works well, there are a number of missing features which would save time of
+our community, for example automatically close PRs from master. Many of them
+are general plugins, which are required by every org, but we need more plugins
+which support coala development workflow.
+
+## New Features
+
+### Auto-assign issue based on commit mention in a PR
+
+When a PR is opened in the repo and if any of it's commit
+mentions(fixes, closes) a unassigned issues, auto assign commit author to the
+mentioned unassigned issues.
+
+### Auto-reject PR opened on master of forks
+
+Normally, contributors should make changes in different branch from master of
+forks, and submit a PR. But, sometimes we receive PRs from master of forks.
+To solve this problem, we need a mechanism by which gitmate automatically close
+those PR’s, and comment on the PR informing the PR author to work on a
+different branch, comment message will be configurable.
+
+### Auto-mark PR with necessary labels as per the reviews done
+
+When a reviewer reviews a PR, he either approves changes or request changes or
+comments something and leaves it there, but the PR would be still in the review
+queue, so there should be an automation to mark approve label if reviewer
+approves changes and mark WIP if reviewer requests changes or comments on the
+source code when there is CI failure. This will be a part of ack plugin.
+A setting to limit this plugin to certain teams will be added.
+
+### Fixes vs Closes automated checking
+
+Our members often confuse when we should use `Fixes` or `Closes` in a commit
+message. So, GitMate should automatically check every commit for issues which
+is being Fixed or Closed, if those issues have bug label (configurable),
+`Fixes` should be used, and vice versa. If there is an error, GitMate will
+comment on the PR with a configurable message and remove 'pending review'
+labels, mark 'wip' label, this will be part of ack plugin.
+
+### Assign Plugin
+
+Assign issue when a user comments on an issue requesting for an `assignment`
+to that issue. Will support settings for each team like limit team to work on
+issues with specific set of labels, limit number of issues of specified label,
```

Comment:
I think the comma at the end of this line is not wanted.
otherwise I dont know what this means.